### PR TITLE
Created new contact UI

### LINKED ISSUE #5: Create a new contact UI

### PR DESCRIPTION
**Pull request by muhsinfazal**
Fixes #5 
- Fixed previous reviews.
- Added new contact pane
- Added contact delete functionality

### CODE REVIEW

File: app/javascript/src/components/Common/DeleteAlert.jsx [hunk 1]
```diff
@@ -2,15 +2,15 @@ import React, { useState } from "react";
 
 import { Alert, Toastr } from "neetoui/v2";
 
-const DeleteAlert = ({ focusedNoteId, onClose, setNotes }) => {
+const DeleteAlert = ({ focusedItemId, onClose, setState }) => {
   const [isDeleting, setIsDeleting] = useState(false);
 
-  const notDeleted = note => note.id !== focusedNoteId;
+  const notDeleted = contact => contact.id !== focusedItemId;
```
Comment (yedhink) [inline app/javascript/src/components/Common/DeleteAlert.jsx:8]: If it's a boolean value(return value or variable value) then prefix with something like `is` or `has` etc. Boolean values denote a question.

File: app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React from "react";
+
+import { Formik, Form } from "formik";
+import { Button, Pane } from "neetoui/v2";
+import { Input, Select } from "neetoui/v2/formik";
+
+import formValidationSchemas from "constants/formValidationSchemas";
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx:7]: Validation schemas should belong to the nearest scope. Having a single `formValidationSchemas` to store all validation schemas is NOT a good pattern.

Create a `./constants.js` and create a `VALIDATION_SCHEMA`. Keep this pattern of keeping things to the closest scope as possible, in your mind!

File: app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import React from "react";
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx:1]: You still haven't figured out the naming patterns even after so many reviews!

```
Contacts/ContactPane/ContactForm.jsx 
```

Figure out the issue in the naming in the above path. You have to reply to this comment on what exactly you think is the issue and then rectify it. If you don't understand what's the issue, then please do ask, and I can help you out.

File: app/javascript/src/constants/formValidationSchemas.js [hunk 4]
```diff
@@ -32,4 +32,12 @@ export default {
       .required("Tags is required")
       .nullable(),
   }),
+  contactsForm: Yup.object().shape({
```
Comment (yedhink) [inline app/javascript/src/constants/formValidationSchemas.js:35]: `Yup` should be lowercase `yup`. What's your reasoning for making this capital `Yup`?

**Review by yedhink — COMMENTED** [review]
@muhsinfazal _a

Take a look into these comments.

File: app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx (hunk 3 above)
Comment (muhsinfazal) [reply app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx:1]: I think the issue is the redundant word "Contact" in the path, instead, it should've been `Contacts/Pane/Form.jsx`.
Am I right?

File: app/javascript/src/constants/formValidationSchemas.js (hunk 4 above)
Comment (muhsinfazal) [reply app/javascript/src/constants/formValidationSchemas.js:35]: It was already being exported as `Yup` in the neeto-ui-challenge template code. So I went along with it.
Will change it to `yup`.

File: app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx (hunk 3 above)
Comment (yedhink) [reply app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx:1]: @muhsinfazal Yes. So you do understand the naming convention. So follow it from next time onwards.

Moving forward I would highly appreciate if you don't repeat this mistake again. Thanks.

File: app/javascript/src/constants/formValidationSchemas.js (hunk 4 above)
Comment (yedhink) [reply app/javascript/src/constants/formValidationSchemas.js:35]: Understood. I will correct it in wheel. Thanks for pointing this out. Appreciate it.

File: app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx (hunk 3 above)
Comment (muhsinfazal) [reply app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx:1]: Initially, I had it this way: `Contacts/Pane/Form.jsx` but then I was not able to name the React component as `Pane` as there was already a `Pane` being imported from NeetoUI inside that file. Same case for `Form` as there was a Form being imported from Formik inside that file.

```
import React from "react";

import { Formik, Form } from "formik";
import { Button, Pane } from "neetoui/v2";
import { Input, Select } from "neetoui/v2/formik";

import formValidationSchemas from "constants/formValidationSchemas";

import { ROLE_OPTIONS } from "../constants";

const ContactForm = ({ onClose, contact, handleSubmit }) => {
  return (
    <Formik
      initialValues={contact}
      onSubmit={handleSubmit}
      validationSchema={formValidationSchemas.contactsForm}
    >
      {({ isSubmitting, handleSubmit }) => (
        <Form className="w-full">
          <Pane.Body>
            <div className="space-y-8 w-full">
```

So, I decided to rename the React Component as well as the filename. 

What was the correct approach deal with this??

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Contacts/ContactPane/ContactForm.jsx:1]: @muhsinfazal Read: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#import_an_export_with_a_more_convenient_alias

**Pull request merged**